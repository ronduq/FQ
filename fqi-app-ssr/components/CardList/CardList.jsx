import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash.get';
import classNames from 'classnames';
import Card from '../Card/CardContainer';
import s from './CardList.scss';


const CardList = ({
  retailerProduces,
  selectedRetailerId,
  retailersData,
  producesData,
}) => {
  return (
    <ul className={classNames('layout', 'layout--3-3-3', s.cardContainer)}>
      {Object.keys(get(retailerProduces, 'items', '')).map((retailerProduce, index) => {
        while (index < 3){
          return (
            <li className='layout__column--1' key={`card${index}`}>
              <Card 
                produce={retailerProduces.items[retailerProduce]} 
                retailerId={selectedRetailerId}
                logo={retailersData.itemsLogos[selectedRetailerId]}
                name={producesData.itemsLabels[retailerProduces.items[retailerProduce].produceCode]}
                image={producesData.itemsImages[retailerProduces.items[retailerProduce].produceCode]}
              />
            </li>
          )
        }
      })}
    </ul>
  );
};

CardList.propTypes = {
  retailerProduces: PropTypes.shape(),
  selectedRetailerId: PropTypes.string,
  retailersData: PropTypes.shape(),
  producesData: PropTypes.shape(),
};

CardList.defaultProps = {
  retailerProduces: {},
  selectedRetailerId: '',
  retailersData: {},
  producesData: {},

};

export default CardList;
