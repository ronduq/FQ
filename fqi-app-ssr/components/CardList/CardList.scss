.cardContainer {
  grid-auto-rows: 1fr;
}