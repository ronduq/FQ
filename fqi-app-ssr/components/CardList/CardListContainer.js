import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import CardList from './CardList';
import { 
  getRetailersState,
  getProducesState,
  getRetailerProduces,
  getSelectedRetailer,
} from '../../selectors/index';

const mapStateToProps = state => {
  return {
    retailersData: getRetailersState(state),
    producesData: getProducesState(state),
    retailerProduces: getRetailerProduces(state),
    selectedRetailerId: getSelectedRetailer(state) ,
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CardList);