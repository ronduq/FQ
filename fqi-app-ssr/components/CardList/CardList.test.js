/* eslint-env jest */

import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import CardList from './CardList'

describe('With Snapshot Testing', () => {
  const handleClick = jest.fn();

  const props = {
    produce: {
      id: "apple-gala",
      countriesOfOrigin: {
        '0': {countryName: "Algeria", isOrganic: true},
        '1': {countryName: "France"}
      },
      perception: "2",
      quality: "98",
      scanDate: "00/01",
      value: "2"
    },
    logo: 'https://cdn.buttercms.com/Bi087WwSQGi55NK09dsA',
    name: 'Gala Apples',
    image: "https://cdn.buttercms.com/mlIKiElrQdeZ8m7syIMx",
    cardLabels: {
      countries_of_origin: "Countries of origin",
      last_scan: "Last scan",
      organic: "Organic",
      perception: "Perception",
      product_of: "Product of",
      profile: "Profile",
      quality: "Quality",
      trends: "Trends",
      value: "Value",
    }

  }
  it('Card snapshot"', () => {
    const component = renderer.create(<CardList />)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  });

})