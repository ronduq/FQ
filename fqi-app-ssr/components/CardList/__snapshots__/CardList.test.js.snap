// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Card snapshot" 1`] = `
<ul
  className="layout layout--3-3-3 cardContainer"
/>
`;
